File: CS/09_Forms/ExtractJavaScript.cs
using Spire.Pdf;
using Spire.Pdf.Actions;
using Spire.Pdf.Fields;
using Spire.Pdf.Widget;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Text;
using System.Windows.Forms;

namespace ExtractJavaScript
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a pdf document
            PdfDocument pdf = new PdfDocument();

            //Load a pdf document
            pdf.LoadFromFile(@"..\..\..\..\..\..\Data\ExtractJavaScript.pdf");

            string js = null;

            PdfFormWidget form = pdf.Form as PdfFormWidget;
            for (int i = 0; i < form.FieldsWidget.List.Count; i++)
            {
                PdfField field = form.FieldsWidget.List[i] as PdfField;
                if (field is PdfTextBoxFieldWidget)
                {
                    PdfTextBoxFieldWidget textbox = field as PdfTextBoxFieldWidget;

                    //Find the textbox named total
                    if (textbox.Name == "total")
                    {
                        //Get the action
                        PdfJavaScriptAction jsa = textbox.Actions.Calculate;

                        if (jsa != null)
                        {
                            //Get JavaScript
                            js = jsa.Script;
                        }                       
                    }
                }
            }

            //Save and launch the result file
            File.WriteAllText("ExtractJavaScript.txt", js);
            System.Diagnostics.Process.Start("ExtractJavaScript.txt");
        }

    }
}

File: VB/09_Forms/ExtractJavaScript.vb
Imports Spire.Pdf
Imports Spire.Pdf.Actions
Imports Spire.Pdf.Fields
Imports Spire.Pdf.Widget
Imports System.IO

Namespace ExtractJavaScript
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PdfDocument object
            Dim pdf As New PdfDocument()

            ' Load the PDF document from the specified file path
            pdf.LoadFromFile("..\..\..\..\..\..\Data\ExtractJavaScript.pdf")

            ' Initialize a variable to store the JavaScript code
            Dim js As String = Nothing

            ' Get the form widget from the loaded PDF document
            Dim form As PdfFormWidget = TryCast(pdf.Form, PdfFormWidget)

            ' Iterate through each field in the form
            For i As Integer = 0 To form.FieldsWidget.List.Count - 1
                ' Get the current field
                Dim field As PdfField = TryCast(form.FieldsWidget.List(i), PdfField)

                ' Check if the field is a TextBox field
                If TypeOf field Is PdfTextBoxFieldWidget Then
                    ' Cast the field as a TextBox field
                    Dim textbox As PdfTextBoxFieldWidget = TryCast(field, PdfTextBoxFieldWidget)

                    ' Find the textbox named "total"
                    If textbox.Name = "total" Then
                        ' Get the Calculate action of the textbox
                        Dim jsa As PdfJavaScriptAction = textbox.Actions.Calculate

                        ' Check if a JavaScript action is associated with the textbox
                        If jsa IsNot Nothing Then
                            ' Get the JavaScript code from the action
                            js = jsa.Script
                        End If
                    End If
                End If
            Next i

            ' Write the JavaScript code to a text file
            File.WriteAllText("ExtractJavaScript.txt", js)

            ' Close the document
            pdf.Close()

            Process.Start("ExtractJavaScript.txt")
        End Sub

    End Class
End Namespace
